src/output.cpp

```diff
@@ -1412,72 +1412,51 @@ std::string from_sentence_case (const std::string &kingston)
     return "";
 }
 
-std::string vstring_format(const char *pattern, va_list argptr)
+#if defined(_MSC_VER)
+std::string vstring_format(char const *const format, va_list args)
 {
-    // If we have no C++11 support, define a hackish way to do va_copy
-    // See http://stackoverflow.com/questions/558223/va-copy-porting-to-visual-c
-    // and http://stackoverflow.com/questions/5047971/how-do-i-check-for-c11-support
-#if __cplusplus < 201103L && !defined(va_copy)
-#define va_copy(dest, source) dest = source
-#endif
+    int const result = _vscprintf_p(format, args);
+    if (result == -1) {
+        return std::string("Bad format string for printf.");
+    }
 
-    int buffer_size = 1024; // Any number is good
-    int returned_length = 0;
-    std::vector<char> buffer(buffer_size, '\0');
-    // Call of vsnprintf() makes va_list unusable, so we need a copy.
-    va_list cur_argptr;
-#if (defined(_WIN32) || defined(WINDOWS) || defined(__WIN32__))
-    // Microsofts vsnprintf does return -1 on buffer overflow, not
-    // the required size of the buffer. So we have to increase the buffer
-    // until we succeed.
-    while(true) {
-        buffer.resize(buffer_size, '\0');
-        va_copy(cur_argptr, argptr);
-        returned_length = vsnprintf(&buffer[0], buffer_size, pattern, cur_argptr);
-        va_end(cur_argptr);
-        if( returned_length >= 0 && returned_length <= buffer_size ) {
-            // Buffer size was sufficient, string has been printed, all is well
+    std::string buffer(result, '\0');
+    _vsprintf_p(&buffer[0], result + 1, format, args); //+1 for string's null
+
+    return buffer;
+}
+#else
+std::string vstring_format(char const *const format, va_list args)
+{
+    errno = 0; // Clear errno before trying
+    std::vector<char> buffer(1024, '\0');
+
+    while (size_t const buf_size = buffer.size()) { // aka while (true)
```

Comment:
If silencing that warning is desirable, I'd prefer something more idiomatic
than sticking a random nearby expression into it.  If the comment was
necessary it should have mentioned the warning, comments should say why,
not how, we can see how.